editor.planx.uk/src/@planx/components/FileUploadAndLabel/Public/SelectMultipleFileTypes.tsx

```diff
@@ -102,73 +54,99 @@ export const SelectMultipleFileTypes = (props: SelectMultipleProps) => {
+    (option: Option) =>
+    (event?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      const isCurrentlyChecked = initialTags.includes(option.name);
+      const newCheckedState = !isCurrentlyChecked;
 
-    switch (reason) {
-      case "selectOption": {
-        const updatedTags = selectedTags.filter(
-          (tag) => !initialTags?.includes(tag),
-        );
+      if (newCheckedState) {
+        // Add the tag
         const updatedFileList = addOrAppendSlots(
-          updatedTags,
+          [option.name],
           uploadedFile,
           fileList,
         );
         setFileList(updatedFileList);
-        break;
-      }
-      case "removeOption": {
-        const removedTags = initialTags?.filter(
-          (tag) => !selectedTags?.includes(tag),
-        );
+      } else {
+        // Remove the tag
         const updatedFileList = removeSlots(
-          removedTags,
+          [option.name],
           uploadedFile,
           fileList,
         );
         setFileList(updatedFileList);
-        break;
       }
-    }
-  };
+    };
 
   return (
-    <SelectMultiple
-      getOptionLabel={(option) => option.name}
-      groupBy={(option) => option.category}
-      id={`select-multiple-file-tags-${uploadedFile.id}`}
-      isOptionEqualToValue={(option, value) => option.name === value.name}
-      key={`form-${uploadedFile.id}`}
-      label={
-        <>
-          What does this file show? (select all that apply)
-          <Box component="span" sx={visuallyHidden}>
-            This question refers to file: {uploadedFile.file.name}
+    <Box
+      sx={(theme) => ({
+        width: "100%",
+        backgroundColor: theme.palette.background.default,
+        borderColor: theme.palette.border.main,
+        borderStyle: "solid",
+        borderWidth: "1px",
+        borderTopColor: theme.palette.border.light,
+        padding: theme.spacing(2),
+      })}
+    >
+      <Typography variant="h3" mb={3} id={titleId}>
+        What does this file show? (select all that apply)
+        <Box component="span" sx={visuallyHidden}>
+          This question refers to file: {uploadedFile.file.name}
+        </Box>
+      </Typography>
+
+      {Object.entries(groupedOptions).map(([category, categoryOptions]) => (
+        <FormControl
+          key={category}
+          component="fieldset"
+          sx={{
+            width: "100%",
+            mb: 1,
+          }}
+          aria-describedby={titleId}
+        >
+          <FormLabel
+            component="legend"
+            sx={{
+              fontWeight: FONT_WEIGHT_SEMI_BOLD,
+              color: "text.primary",
+            }}
+          >
+            {capitalize(category)} information
+          </FormLabel>
+
+          <Box>
+            {categoryOptions.map((option) => (
+              <ChecklistItem
+                key={`${category}-${option.name}`}
+                id={`${sanitizeId(uploadedFile.id)}-${sanitizeId(category)}-${sanitizeId(option.name)}`}
```

Comment:
```suggestion
                id={sanitizeId(`${uploadedFile.id}-${category}-${option.name}`)}
```

I think this would output the same value and is a bit more readable?